### DISCUSSION TITLE
llama.cpp port to pure Golang

### DISCUSSION THREAD

**Discussion by gotzmann** [ggml-org/llama.cpp#919] [Show and tell]
I've released [the framework](https://github.com/gotzmann/llama.go) I have been building for the last month. 

It's the like 1:1 port of llama.cpp to pure Go, where the language really shines due to easier multi-threading model and channels.

Not so performant and memory savvy as original one but I hope it would be useful for those who'd like to grok with ChatGPT-like projects with Go:)

**Comment by ggerganov** [comment]
Congrats! Looks like great effort and very well done!

**Comment by golangpkg** [comment]
pure golang is create ! golang is easy to use .

**Comment by sorasoras** [comment]
@gotzmann 
The perf might be close if using cublas and roblas.
what do you think
